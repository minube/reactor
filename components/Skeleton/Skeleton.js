import { number } from 'prop-types';
import { View } from 'react-native';
import React, { Component } from 'react';
import styles from './Skeleton.style';

import Motion from '../Motion';
import { THEME } from '../../common';

const { MOTION: { DURATION } } = THEME;

export default class Skeleton extends Component {
  static propTypes = {
    opacity: number,
  };

  static defaultProps = {
    opacity: 0.5,
  };

  state = { visible: false };

  interval = undefined;

  componentDidMount() {
    this.interval = setInterval(() => {
      const { state: { visible } } = this;
      this.setState({ visible: !visible });
    }, DURATION * 3);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    const { props: { opacity, ...inherit }, state: { visible } } = this;

    return (
      <View style={[styles.container, inherit.style]}>
        <Motion
          duration={DURATION * 3}
          style={styles.motion}
          timeline={[{ property: 'opacity', value: visible ? opacity : 0 }]}
        />
      </View>
    );
  }
}
