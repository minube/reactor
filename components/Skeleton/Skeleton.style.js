import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const { COLOR, UNIT } = THEME;

export default StyleSheet.create({
  container: {
    backgroundColor: COLOR.BASE,
    minHeight: UNIT,
    minWidth: UNIT,
  },

  motion: {
    backgroundColor: COLOR.WHITE,
    width: '100%',
    height: '100%',
  },
});
