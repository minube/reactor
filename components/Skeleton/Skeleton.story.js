import React from 'react';
import { storiesOf } from '@storybook/react';
import { number, object } from '@storybook/addon-knobs/react';

import Skeleton from './Skeleton';

const STYLE = {
  height: 128,
  width: 128,
};

storiesOf('✅ Skeleton', module)
  .add('default', () => (
    <Skeleton />
  ))
  .add('opacity', () => (
    <Skeleton opacity={1} />
  ))
  .add('style', () => (
    <Skeleton style={STYLE} />
  ))
  .add('🏀 Playground', () => (
    <Skeleton
      opacity={number('opacity', 0.5)}
      style={object('style', STYLE)}
    />
  ));
