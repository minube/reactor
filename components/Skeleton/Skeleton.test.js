import React from 'react';
import renderer from 'react-test-renderer';

import Skeleton from './Skeleton';

const STYLE = {
  height: 128,
  width: 128,
};

describe('<Skeleton>', () => {
  it('renders', () => {
    const tree = renderer.create(<Skeleton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {opacity}', () => {
    const tree = renderer.create(<Skeleton opacity={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Skeleton style={STYLE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
