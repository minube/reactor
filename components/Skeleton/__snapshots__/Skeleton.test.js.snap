// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton> renders 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#cdd2d4",
        "minHeight": 10,
        "minWidth": 10,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "height": "100%",
        "opacity": 0,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`<Skeleton> when {opacity} 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#cdd2d4",
        "minHeight": 10,
        "minWidth": 10,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "height": "100%",
        "opacity": 0,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`<Skeleton> when {style} 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#cdd2d4",
        "minHeight": 10,
        "minWidth": 10,
      },
      Object {
        "height": 128,
        "width": 128,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "height": "100%",
        "opacity": 0,
        "width": "100%",
      }
    }
  />
</View>
`;
